### PR TITLE
LSP: Do not compute text edits in source actions on `resolve` call.

### PR DESCRIPTION
**Pull request by dbalek**
Modified to follow change in VSCode behavior when `codeAction/resolve` LSP call is invoked also on mouse hover and not only on source action selection (see https://github.com/microsoft/vscode/issues/175008). The source actions that require user interaction to compute their workspace edits should no more do it on `codeAction/resolve`.

### CODE REVIEW

**Review by MartinBalin — APPROVED** [review]

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/EqualsHashCodeGenerator.java [hunk 1]
```diff
@@ -162,11 +173,10 @@ public CompletableFuture<CodeAction> resolve(NbCodeLanguageClient client, CodeAc
                                 org.netbeans.modules.java.editor.codegen.EqualsHashCodeGenerator.generateEqualsAndHashCode(wc, tp, generateEquals ? selectedFields : null, generateHashCode ? selectedFields : null, -1);
                             }
                         });
-                        if (!edits.isEmpty()) {
-                            codeAction.setEdit(new WorkspaceEdit(Collections.singletonMap(uri, edits)));
-                        }
+                        future.complete(edits.isEmpty() ? null : new WorkspaceEdit(Collections.singletonMap(uri, edits)));
```
Comment (sdedic) [inline java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/EqualsHashCodeGenerator.java:176]: Would be great if the lsp.api > lsp4j bridge (`Util.workspaceEditFromApi`) worked even here, for `CodeActionProvider`'s commands.

**Review by sdedic — APPROVED** [review]

File: java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/EqualsHashCodeGenerator.java (hunk 1 above)
Comment (dbalek) [reply java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/EqualsHashCodeGenerator.java:176]: The `CodeActionsProvider` implementations from the `java.lsp.server` module do not use the `lsp.api` data structures (`lsp4j` data structures are used directly there). There is no need for using bridge...

**Comment by MartinBalin** [comment]
This has to go to NB21 as when using mouse, like those of us who don't remember 3keys shortcut on mac then this always triggers the action when hoovering over menu item. Ugly behaviour now in VSCode.

**Comment by dbalek** [comment]
Unfortunately, simple rebasing would bring a lot of unwanted commits to delivery. Merging to master.

**Comment by neilcsmith-net** [comment]
@dbalek so you're saying this change cannot go into NB21?

**Comment by MartinBalin** [comment]
Exactly @neilcsmith-net , too difficult to make it to NB21.

**Comment by neilcsmith-net** [comment]
Thanks @MartinBalin There are two PRs (this and #6996 ) that you've said must go in NB21 and then have been merged to master without confirmation from you in review.  Just checking we're not expecting anything related to either to hit delivery?

**Pull request merged**